vivarium/controllers/notebook_controller.py

```diff
@@ -291,6 +291,13 @@ def has_eaten(self):
         val = self.ate
         self.ate = False
         return val
+
+    def has_eaten_since(self, time):
+        """ Check if the agent has eaten since a given time
+
+        :return: True if the agent has eaten since the given time, False otherwise
+        """
+        return self.time_since_last_meal < time
```

Comment:
Double check this will work as expected (init of the variable, comparison operator..)